cms/models/pluginmodel.py

```diff
@@ -250,25 +251,23 @@ def save(self, no_signals=False, *args, **kwargs):
     def reload(self):
         return CMSPlugin.objects.get(pk=self.pk)
 
+    @transaction.atomic
     def move(self, target, pos=None):
+        from cms.utils.plugins import reorder_plugins
         super(CMSPlugin, self).move(target, pos)
         self = self.reload()
-        try:
-            new_pos = max(CMSPlugin.objects.filter(parent_id=self.parent_id,
-                                                   placeholder_id=self.placeholder_id,
-                                                   language=self.language).exclude(pk=self.pk).order_by('depth', 'path').values_list('position', flat=True)) + 1
-        except ValueError:
-            # This is the first plugin in the set
-            new_pos = 0
-        self.position = new_pos
-        self.save()
+        plugins = CMSPlugin.objects.filter(
+            parent_id=self.parent_id, placeholder_id=self.placeholder_id, language=self.language
+        ).order_by('path').values_list('pk', flat=True)
+        reorder_plugins(self.placeholder, self.parent_id, self.language, plugins)
```

Comment:
On the move endpoint, we already to a calculation of positions so this would be redundant.
I think the logic should be here but then we should add an order parameter to ``move()`` or stick with the utility to not touch an internal of treebeard.
Also ordering by path and then supplying that as order can backfire because they can be out of sync.